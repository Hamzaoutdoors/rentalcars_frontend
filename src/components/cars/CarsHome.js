import React from 'react';
import styled from 'styled-components';
import { useSelector } from 'react-redux';
import Slider from '../Slider';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 100vw;
  max-width: 1320px;
  height: 100vh;
  color: black;
  justify-content: center;
  align-items: center;
  overflow-x: hidden;
`;

const Title = styled.h1`
  font-size: 3em;
  text-align: center;
  font-weight: bold;
  text-shadow: 2px 2px #000;
  padding: 1rem;
`;

const SubTitle = styled.h2`
  font-size: 0.8em;
  text-align: center;
  letter-spacing: 1px;
  padding-bottom: 20px;
  font-weight: bold;
  color: gray;
  border-bottom: 2px dotted #D3D3D3;

`;

const HomePage = () => {
  const { isAuthenticated } = useSelector((state) => state.auth);

  return (
    <Container>
      {
        isAuthenticated
          ? (
            <>
              <Title> LATEST MODELS</Title>
              <SubTitle>Please select your car</SubTitle>
              <Slider />
            </>
          ) : (
            <>
              <Title>Welcome to Car Rental App</Title>
              <SubTitle>Please sign in to explore more</SubTitle>
              <Slider />
            </>
          )
      }

    </Container>
  );
};

export default HomePage;
