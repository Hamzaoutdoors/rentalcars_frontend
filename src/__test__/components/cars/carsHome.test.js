import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import CarsHome from '../../../components/cars/CarDetails';

it('CarsHome component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <CarsHome />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
