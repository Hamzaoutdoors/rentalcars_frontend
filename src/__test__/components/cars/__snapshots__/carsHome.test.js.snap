// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarsHome component renders correctly 1`] = `
<div
  className="sc-jrQzAO gURVnT"
/>
`;
